src/main/java/com/minecolonies/blockout/controls/ButtonHandler.java

```diff
@@ -0,0 +1,15 @@
+package com.minecolonies.blockout.controls;
+
+/**
+ * Used for windows that have buttons and want to respond to clicks.
+ */
+@FunctionalInterface
+public interface ButtonHandler
```

Comment:
I had originally ignored this, because removing the interface and using Consumer would mean things like this `public class WindowHireWorker extends Window implements ButtonHandler`, would implement consumer, and have some random `accept` method which wouldn't nearly be as readable as `onButtonClicked`. It looks like I can _fix_ this by extending my interface from Consumer<Button> and adding some boilerplate code:
```
@FunctionalInterface
public interface ButtonHandler extends Consumer<Button>
{
    /**
     * Called when a button is clicked.
     *
     * @param button the button that was clicked.
     */
    void onButtonClicked(Button button);

    default void accept(Button button) {
        onButtonClicked(button);
    }
}
```